schema/workflow.json

```diff
@@ -1,33 +1,26 @@
 {
-  "$id": "https://serverlessworkflow.io/schemas/0.8/workflow.json",
+  "$id": "https://serverlessworkflow.io/schemas/0.9/workflow.json",
   "$schema": "http://json-schema.org/draft-07/schema#",
   "description": "Serverless Workflow specification - workflow schema",
   "type": "object",
   "properties": {
-    "id": {
-      "type": "string",
-      "description": "Workflow unique identifier",
-      "minLength": 1
-    },
-    "key": {
+    "name": {
```

Comment:
Shouldn't the definition's name follow workflow [naming conventions](https://github.com/serverlessworkflow/specification/blob/main/specification.md#naming-convention)?

> Specifically, names must be lowercase, start and end with an alphanumeric character, and consist entirely of alphanumeric characters with optional isolated medial dashes '-' (i.e., dashes must not be adjacent to each other).

I personally would advocate enforcing it, especially now that it's to become (part of) an identifier, and would make the spec more consistent (as we enforce it for all other `name` properties).

If so, there's the need to add a pattern property, and possibly update the description.

Reply:
ok, to be honest I did not realize the convention was not being enforced because of the lack of regular expression in the schema

Reply:
Yeah, no worries. I think it was voluntarily overlooked precisly because we though of addressing it in this PR 😆